sections/design/introduction-to-react.tex

```diff
@@ -60,48 +62,117 @@ \section{Introduction to React}\label{sec:intro-to-react}
+We can set the initial state of a component by assigning \texttt{this.state} in the constructor of the component.
 \begin{center}
-    \texttt{this.state = \{date = new Date();\};}
+    \texttt{this.state = \{textToDisplay: 'Initial text'\};}
 \end{center}
 The state can then be changed by reassigning to the \texttt{date} variable.
 This is done by using the \texttt{this.setState()} function.
 \begin{center}
-    \texttt{this.setState(\{date = new Date();\});}
+    \texttt{this.setState(\{textToDisplay: 'New text'\});}
 \end{center}
-When a component has a state set it is considered to be a stateful component.
 Parent or child components should not care if a certain component is stateful or stateless since the state is local and not accessible by other components.
-A component can however pass its state through \texttt{props} to a child component, which means that information about a state can only be passed in a \textit{top-down} approach \cite{ReactJS}.
+A component can however pass variables from its state through \texttt{props} to a child component, which means that information about a state can only be passed in a \textit{top-down} approach \cite{ReactJS}.
+Every time a state is changed with \texttt{this.setState()} the component that the state belongs to and all of its children are re-rendered.
+By passing callback to the props of children we can allow  the children of a component to transfer data back to its parent and even trigger re-renders by passing callback functions that set the state of the parent component.
+If a function calls \texttt{this.setState()} it has to be bound in the constructor of the component.
+An example of how this is done can be seen in the following listing.
+\begin{lstlisting}
+class ourComponent extends Component {
+	constructor(props){
+		super(props);
+		this.state = {
+			textToDisplay: 'initial text'
+		};
+
+		this.handleChange = this.handleChange.bind(this);
+	}
+
+	handleChange(text){
+		this.setState({
+			textToDisplay: text
+		});
+	}
+
+	render(){
+		return <ChildComponent handleChange={this.handleChange} />
+	}
+}
+\end{lstlisting}
+
+\subsection*{Lifecycle}
+When a component is implemented as a class that extends \texttt{Component} from the react library, it gives us access to some functions that are automatically called at different points in the lifecycle of a component.
+The function that we will mostly be using is \texttt{componentDidMount}.
+This function is called right after the component is rendered for the first time in its lifecycle.
+This is useful for when we need to do fetch data from the API that we want to save in the state of the component.
+The reason that this is not done in the constructor of the component is because, the render function should not be waiting for asynchronous operations before it is allowed to render.
+This is because we do not know how long it will take before function is done executing or if it will even succeed.
```

Comment:
```suggestion
This is because we do not know how long it will take before the asynchronous function is done executing, or if it will even succeed.
```